### DISCUSSION TITLE
Condition rendering (channelMap) inside codeTemplate function

### DISCUSSION THREAD

**Discussion by kyulol** [nextgenhealthcare/connect#5740] [Q&A]
Hello everyone,
I'm a begginner with Mirth.
I actually encounter a problem using a function with inside a if/else statement.
The idea is to check if the condition is true and add the channelMap and if false abort sending finalResult to another channel.

`function testCondition() {
  var finalResult;
  finalResult =  'some text';
	if(1 == 1){
                channelMap.put('finalResult', finalResult);
	} else {
		destinationSet.removeAll();
	}
}`

my tests let me confuse because it work very well without a function :

`  var finalResult;
  finalResult =  'some text';
	if(1 == 1){
                channelMap.put('finalResult', finalResult);
	} else {
		destinationSet.removeAll();
	}`

Thanks for your help =)

**Comment by joaryche** [comment]
you have to remember to run the function.. how is this handled?
